src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    private Object[] grow() {
+        int oldCapacity = elementData.length;
+        Object[] newElementData = new Object[oldCapacity + (oldCapacity >> 1)];
+        System.arraycopy(elementData, 0, newElementData, 0, oldCapacity);
+        return newElementData;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < elementData.length) {
+            this.elementData[size] = value;
+            size++;
+        } else {
+            elementData = grow();
+            elementData[size] = value;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index outs of bounds");
+        }
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        this.elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index outs of bounds");
+        }
```

Comment:
you have the same check in several places, create a separate method for this logic and add a more informative message